galaxy_ng/app/api/ui/serializers/synclist.py

```diff
@@ -45,23 +74,108 @@ def _sanitize_groups(self, group_names):
 
         return groups
 
+    def _get_repository(self, repository_id):
+        try:
+            repository = AnsibleRepository.objects.get(pulp_id=repository_id)
+            # TODO: validate there is a distro associatd?
+            return repository
+        except AnsibleRepository.DoesNotExist:
+            errmsg = \
+                'Repository "{pulp_id}" not found while creating synclist'
+            raise ValidationError(
+                errmsg.format(pulp_id=repository_id))
+
     def to_internal_value(self, data):
-        groups = data.get('groups')
+        groups = data.get('groups', [])
         data['groups'] = self._sanitize_groups(groups)
+
+        users = data.get('users', [])
+        data['users'] = self._sanitize_users(users)
+
+        repository_data = data.get('repository', None)
+        if repository_data:
+            data['repository'] = self._get_repository(repository_data)
+
+        upstream_repository_data = data.get('upstrean_repository', None)
+        if upstream_repository_data:
+            data['upstream_repository'] = self._get_repository(upstream_repository_data)
+        else:
+            # If not specified, use the default upstream repo
+            data['upstream_repository'] = \
+                AnsibleRepository.objects.get(name=default_repo_name)
         return super().to_internal_value(data)
 
+    def _name_slug_from_group_names(self, groups_data):
+        ngs = auth_models.Group.objects.filter(name__in=[x.name for x in groups_data])
+        ngs = ngs.filter(name__ne=auth_models.RH_PARTNER_ENGINEER_GROUP)
+
+        if not ngs.filter():
+            # TODO: better error
+            errmsg = 'No Group found while creating synclist'
+            raise ValidationError(errmsg)
+
+        # FIXME: further filter for group name scope?
+
+        account_name = ngs.filter()[0].account_number()
+        name_slug = settings.GALAXY_API_SYNCLIST_NAME_FORMAT.format(account_name=account_name)
+
+        return name_slug, account_name
+
+    @transaction.atomic
+    def _create_repository(self, name_slug, account_name):
+        # lookup groups, to get a group name to use in naming the repository/distro
+        try:
+            description = f"Synclist repository for account {account_name}"
+
+            repository, created = AnsibleRepository.objects.get_or_create(name=name_slug)
+            repository.description = description
+            repository.save()
+            return repository
+        except IntegrityError as exc:
+            raise ValidationError("AnsibleRepository %s already exists: %s" % (name_slug, exc))
+        except Exception as exc:
```

Comment:
Why are you handling generic exception here for nothing than logging?
This exception must be handled by the top-level exception handler.